### PR TITLE
RD-5755 Part 4 move `app/containers/status` to `app/components/status`

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Fourth PR. In this PR, I  move files from `app/containers/status` to `app/components/status`. Jira ticket: RD-5755

## Screenshots / Videos
N/A

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

✅ Passed - 7-DEC-2022 13:11 IL time `test/cypress/integration/cluster_status_spec.ts` and `test/cypress/integration/widgets/cluster_status_spec.ts`:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4045/pipeline

✅ Passed - `test/cypress/integration/cluster_status_spec.ts` and `test/cypress/integration/widgets/cluster_status_spec.ts`:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4027/pipeline

 

## Documentation
N/A

### CODE REVIEW

**Comment by leon-good-life** [comment]
I'm trying to fix unit test:
```
 FAIL  test/jest/components/SystemStatusIcon.test.tsx
  (Component) SystemStatusIcon
    ✕ renders with available status (69 ms)
    ✕ renders with unavailable status (4 ms)
    ✕ renders with no status (3 ms)
    ✕ renders with degraded status (3 ms)

  ● (Component) SystemStatusIcon › renders with available status

    Invariant Violation: Could not find "store" in the context of "Connect(SystemStatusIcon)". Either wrap the root component in a <Provider>, or pass a custom React context provider to <Provider> and the corresponding React context consumer to Connect(SystemStatusIcon) in connect options.

      15 |         systemStatus = ClusterServiceStatus.OK;
      16 |
    > 17 |         wrapper = mount(<SystemStatusIcon systemStatus={systemStatus} />);
         |                   ^
      18 |     });
      19 |
      20 |     it('renders with available status', () => {

      at invariant (node_modules/invariant/invariant.js:40:15)
      at ConnectFunction (node_modules/react-redux/lib/components/connectAdvanced.js:150:33)
      at renderWithHooks (node_modules/react-dom/cjs/react-dom.development.js:16241:18)
      at updateFunctionComponent (node_modules/react-dom/cjs/react-dom.development.js:18328:20)
      at updateSimpleMemoComponent (node_modules/react-dom/cjs/react-dom.development.js:18266:10)
      at updateMemoComponent (node_modules/react-dom/cjs/react-dom.development.js:18169:14)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:20223:16)
      at HTMLUnknownElement.callCallback (node_modules/react-dom/cjs/react-dom.development.js:336:14)
      at HTMLUnknownElement.callTheUserObjectsOperation (node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
      at innerInvokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:350:25)
      at invokeEventListeners (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:286:3)
      at HTMLUnknownElementImpl._dispatch (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:233:9)
      at HTMLUnknownElementImpl.dispatchEvent (node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:104:17)
      at HTMLUnknownElement.dispatchEvent (node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:241:34)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:385:16)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:440:31)
      at beginWork$$1 (node_modules/react-dom/cjs/react-dom.development.js:25738:7)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:24665:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:24638:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:24237:11)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:23665:7)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:27061:3)
      at node_modules/react-dom/cjs/react-dom.development.js:27485:7
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:24400:12)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:27484:5)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:27572:12)
      at render (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:430:26)
      at fn (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:347:37)
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:24353:12)
      at Object.act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:1090:14)
      at act (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:347:13)
      at Object.wrapAct [as render] (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:416:16)
      at new render (node_modules/enzyme/src/ReactWrapper.js:115:16)
      at mount (node_modules/enzyme/src/mount.js:10:10)
      at Object.<anonymous> (test/jest/components/SystemStatusIcon.test.tsx:17:19)
          at runMicrotasks (<anonymous>)
```

File: app/components/status/SystemServicesStatus.tsx [hunk 1]
```diff
@@ -43,3 +46,21 @@ export default function SystemServicesStatus({
         />
     );
 }
+
+const mapStateToProps = (state: ReduxState) => {
+    return {
+        services: state.manager.clusterStatus.services,
+        isFetching: state.manager.clusterStatus.isFetching,
+        fetchingError: state.manager.clusterStatus.error
+    };
```
Comment (qooban) [inline app/components/status/SystemServicesStatus.tsx:55]: WDYT about this version:
```suggestion
    const { services, isFetching, error: fetchingError } = state.manager.clusterStatus;
    return {
        services,
        isFetching,
        fetchingError
    };
```
?

File: app/components/status/SystemStatusHeader.tsx [hunk 2]
```diff
@@ -28,3 +33,21 @@ export default function SystemStatusHeader({ onStatusRefresh, isFetching }: Syst
         </div>
     );
 }
+
+const mapStateToProps = (state: ReduxState) => {
+    return {
+        isFetching: state.manager.clusterStatus.isFetching,
+        fetchingError: state.manager.clusterStatus.error
+    };
```
Comment (qooban) [inline app/components/status/SystemStatusHeader.tsx:41]: Optional:

See me previous comment about destructuring `state.manager.clusterStatus`. Maybe worth to apply it here as well?

File: app/components/status/SystemStatusIcon.tsx [hunk 3]
```diff
@@ -7,6 +9,19 @@ export interface SystemStatusIconProps {
...
+    const systemStatus = state.manager.clusterStatus.status;
+    return {
+        systemStatus,
+        maintenanceStatus: state.manager.maintenance
+    };
+};
+
+const mapDispatchToProps = () => {
+    return {};
+};
```
Comment (qooban) [inline app/components/status/SystemStatusIcon.tsx:25]: Probably we can shorten it to:
```suggestion
const mapDispatchToProps = () => {};
```

File: app/components/status/SystemServicesStatus.tsx (hunk 1 above)
Comment (leon-good-life) [reply app/components/status/SystemServicesStatus.tsx]: [451c5ac](https://github.com/cloudify-cosmo/cloudify-stage/pull/2310/commits/451c5ac140e05d704226f21113e4fd140c24b263)

File: app/components/status/SystemStatusHeader.tsx (hunk 2 above)
Comment (leon-good-life) [reply app/components/status/SystemStatusHeader.tsx]: [451c5ac](https://github.com/cloudify-cosmo/cloudify-stage/pull/2310/commits/451c5ac140e05d704226f21113e4fd140c24b263)

**Comment by qooban** [comment]
> General: I think the preferable way to do such refactor is to use hooks instead of the `connect` API.

Good point. In case of functional components it's definitely the preferred way.
@leon-good-life Let's refactor it to use `useSelector` and `useDispatch` hooks.

**Comment by leon-good-life** [comment]
I refactored to `useSelector` and `useDispatch` hooks.

**Comment by leon-good-life** [comment]
I fixed the test file and migrated it to TypeScript.

**Review by kubama — DISMISSED** [review]

**Review by qooban — DISMISSED** [review]

**Comment by leon-good-life** [comment]
I fixed lint (import type). So I asked for re-review to be able to merge.
I also verified that tests and checks are passing including system tests after changes.

**Pull request merged**